### PR TITLE
Remove undo steps from everywhere - Closes #5274

### LINKED ISSUE #5274: Remove undo steps from everywhere
### Description

Remove undo steps from:

- Framework: processor / block processor / base block processor
- `lisk-chain`: undo
- `lisk-transactions`: BaseTransaction + each dependent transaction
- `lisk-dpos`: undo
- `lisk-bft`: deleteBlocks

### Motivation

Explicit undo steps are no longer required.

### Acceptance Criteria

- Remove tests for `undo` functionality

### Additional Information

Related issues: #5232

### PR DESCRIPTION
**Pull request by ManuGowda**
### What was the problem?

This PR resolves #5274 

### How was it solved?

Removed undo steps from:

Framework: processor / block processor / base block processor
lisk-chain: undo
lisk-transactions: BaseTransaction + each dependent transaction
lisk-dpos: undo
lisk-bft: deleteBlocks

### How was it tested?

- Removed test and build should be green

### CODE REVIEW

**Comment by ishantiw** [comment]
I think you need to remove undo related tests here as well
https://github.com/LiskHQ/lisk-sdk/blob/5274-remove-undo/elements/lisk-chain/test/unit/process.spec.ts

File: elements/lisk-transactions/src/15_proof_of_misbehavior_transaction.ts [hunk 1]
```diff
@@ -370,42 +370,6 @@ export class ProofOfMisbehaviorTransaction extends BaseTransaction {
 		return errors;
 	}
 
-	protected async undoAsset(
```
Comment (shuse2) [inline elements/lisk-transactions/src/15_proof_of_misbehavior_transaction.ts:373]: We can also remove `asset.reward` property for PoM tx

File: elements/lisk-dpos/src/delegates_info.ts [hunk 2]
```diff
@@ -69,21 +69,6 @@ export class DelegatesInfo {
 		return this._update(header, stateStore, { undo, delegateListRoundOffset });
 	}
 
-	public async undo(
-		header: BlockHeader,
-		stateStore: StateStore,
-		{ delegateListRoundOffset }: DPoSProcessingOptions,
```
Comment (ishantiw) [inline elements/lisk-dpos/src/delegates_info.ts:75]: do we need `DPoSProcessingOptions` type that has `undo` field? maybe we can remove that too

File: elements/lisk-transactions/README.md [hunk 3]
```diff
@@ -56,12 +55,6 @@ All of the abstract methods and properties on the base transaction's interface a
 
 The hallmark of a transaction. Override this static parameter with any number, keeping in mind that the first 10 types (0-9) are reserved for the default transactions.
 
-##### prepare
-
-> prepare(store: StateStorePrepare): Promise<void>
-
-Prepare the relevant information about the accounts, which will be accessible in the later steps during the `apply` and `undo` steps.
-
```
Comment (ishantiw) [inline elements/lisk-transactions/README.md:64]: we still apply so maybe just remove `undo` from the steps

**Pull request merged**